tableaudocumentapi/datasource.py

```diff
@@ -116,6 +127,20 @@ def from_file(cls, filename):
         dsxml = xml_open(filename, cls.__name__.lower()).getroot()
         return cls(dsxml, filename)
 
+    @classmethod
+    def from_scratch(cls, caption, connections):
+        root = ET.Element('datasource', caption=caption, version='10.0')
```

Comment:
I feel like we are going to do this pattern a bunch.  Would it make more sense to build up a logical model that we can then automatically serialize into the xml in some way rather than hard coding specific xml manipulations?